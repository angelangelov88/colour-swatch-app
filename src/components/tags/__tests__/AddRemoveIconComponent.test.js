import renderer from 'react-test-renderer'
import {AddRemoveIconComponent} from "../AddRemoveIconComponent"

it('should render a large size green remove icon', () => {
  const component = renderer.create(
  <AddRemoveIconComponent 
    icon="remove"
    color="green"
    iconSize="lg"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render a small size red add icon', () => {
  const component = renderer.create(
  <AddRemoveIconComponent 
    icon="add"
    color="red"
    iconSize="sm"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render a large size remove icon with gray color', () => {
  const component = renderer.create(
  <AddRemoveIconComponent 
    icon="remove"
    color="gray"
    iconSize="lg"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render a small size primary color add icon', () => {
  const component = renderer.create(
  <AddRemoveIconComponent 
    icon="add"
    color="primary"
    iconSize="sm"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render a large size teal color icon with default icon (which should be remove)', () => {
  const component = renderer.create(
  <AddRemoveIconComponent 
    color="teal"
    iconSize="lg"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
