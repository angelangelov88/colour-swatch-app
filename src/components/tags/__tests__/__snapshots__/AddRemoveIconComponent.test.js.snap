// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a large size green remove icon 1`] = `
<div
  className="font-ibm flex items-center justify-center rounded-xl hover:cursor-pointer w-5 h-5 p-1 text-green-60 hover:text-green-80 hover:bg-green-30 active:text-green-10 active:bg-green-50"
>
  <svg
    aria-hidden="true"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18L18 6M6 6l12 12"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;

exports[`should render a large size remove icon with gray color 1`] = `
<div
  className="font-ibm flex items-center justify-center rounded-xl hover:cursor-pointer w-5 h-5 p-1 text-gray-60 hover:text-gray-80 hover:bg-gray-30 active:text-gray-10 active:bg-gray-50"
>
  <svg
    aria-hidden="true"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18L18 6M6 6l12 12"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;

exports[`should render a large size teal color icon with default icon (which should be remove) 1`] = `
<div
  className="font-ibm flex items-center justify-center rounded-xl hover:cursor-pointer w-5 h-5 p-1 text-teal-60 hover:text-teal-80 hover:bg-teal-30 active:text-teal-10 active:bg-teal-50"
>
  <svg
    aria-hidden="true"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18L18 6M6 6l12 12"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;

exports[`should render a small size primary color add icon 1`] = `
<div
  className="font-ibm flex items-center justify-center rounded-xl hover:cursor-pointer w-4 h-4 p-0.5 text-primary-60 hover:text-primary-80 hover:bg-primary-30 active:text-primary-10 active:bg-primary-50"
>
  <svg
    aria-hidden="true"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 4v16m8-8H4"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;

exports[`should render a small size red add icon 1`] = `
<div
  className="font-ibm flex items-center justify-center rounded-xl hover:cursor-pointer w-4 h-4 p-0.5 text-red-60 hover:text-red-80 hover:bg-red-30 active:text-red-10 active:bg-red-50"
>
  <svg
    aria-hidden="true"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 4v16m8-8H4"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;
